4.x: `integrations.cdi.jpa.TestAnnotationRewriting` test failure

Test `io.helidon.integrations.cdi.jpa.TestAnnotationRewriting` has started failing on the 4.x validation pipeline. Note that the validation workflow seems to be running slowly at the moment which is triggering timing issues in other tests.  First time I've seen this test fail. It passes when run locally as part of the module build (with other tests) but fails when run in isolation (see @spericas comment below).

```
18:29:01,388 [INFO] Running io.helidon.integrations.cdi.jpa.TestAnnotationRewriting
Error: 3,536 [ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.138 s <<< FAILURE! - in io.helidon.integrations.cdi.jpa.TestAnnotationRewriting
Error: 3,537 [ERROR] io.helidon.integrations.cdi.jpa.TestAnnotationRewriting.testNonTransactionalEntityManager  Time elapsed: 0.23 s  <<< ERROR!
org.jboss.weld.exceptions.UnsatisfiedResolutionException: 
WELD-001334: Unsatisfied dependencies for type EntityManager with qualifiers @ContainerManaged @JpaTransactionScoped WELD-001475: The following beans match by type, but none have matching qualifiers:
  - Configurator Bean [class io.helidon.integrations.cdi.jpa.PersistenceExtension, types: JtaExtendedEntityManager, EntityManager, Object, AutoCloseable, DelegatingEntityManager, qualifiers: @Any @Extended @Synchronized @ContainerManaged @Named],
  - Configurator Bean [class io.helidon.integrations.cdi.jpa.PersistenceExtension, types: EntityManager, Object, AutoCloseable, JtaEntityManager, DelegatingEntityManager, qualifiers: @Any @ContainerManaged @JtaTransactionScoped @Synchronized @Named]
  - at org.jboss.weld.bean.builtin.InstanceImpl.checkBeanResolved(InstanceImpl.java:251)
	at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:113)
	at io.helidon.integrations.cdi.jpa.TestAnnotationRewriting.testNonTransactionalEntityManager(TestAnnotationRewriting.java:165)
  ```

I was able to reproduce the issue locally if I run the test in isolation:

```
(~/Projects/helidon-4/integrations/cdi/jpa-cdi) >> mvn clean test -Dtest=TestAnnotationRewriting

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.helidon.integrations.cdi.jpa.TestAnnotationRewriting
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.327 s <<< FAILURE! - in io.helidon.integrations.cdi.jpa.TestAnnotationRewriting
[ERROR] io.helidon.integrations.cdi.jpa.TestAnnotationRewriting.testNonTransactionalEntityManager  Time elapsed: 0.245 s  <<< ERROR!
org.jboss.weld.exceptions.UnsatisfiedResolutionException: 
WELD-001334: Unsatisfied dependencies for type EntityManager with qualifiers @ContainerManaged @JpaTransactionScoped WELD-001475: The following beans match by type, but none have matching qualifiers:
  - Configurator Bean [class io.helidon.integrations.cdi.jpa.PersistenceExtension, types: EntityManager, AutoCloseable, DelegatingEntityManager, Object, JtaExtendedEntityManager, qualifiers: @Any @ContainerManaged @Synchronized @Extended @Named],
  - Configurator Bean [class io.helidon.integrations.cdi.jpa.PersistenceExtension, types: EntityManager, AutoCloseable, DelegatingEntityManager, JtaEntityManager, Object, qualifiers: @Any @ContainerManaged @JtaTransactionScoped @Synchronized @Named]

        at org.jboss.weld.bean.builtin.InstanceImpl.checkBeanResolved(InstanceImpl.java:251)
        at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:113)
        at io.helidon.integrations.cdi.jpa.TestAnnotationRewriting.testNonTransactionalEntityManager(TestAnnotationRewriting.java:165)
```

--- a/integrations/cdi/jpa-cdi/src/test/java/io/helidon/integrations/cdi/jpa/TestAnnotationRewriting.java
+++ b/integrations/cdi/jpa-cdi/src/test/java/io/helidon/integrations/cdi/jpa/TestAnnotationRewriting.java
@@ -39,7 +39,6 @@ import jakarta.transaction.TransactionManager;
 import jakarta.transaction.Transactional;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
@@ -58,40 +57,61 @@ import static org.junit.jupiter.api.Assertions.fail;
         "user=sa"
     }
 )
+// Note that this class tests JpaExtension behavior, not PersistenceExtension behavior. JpaExtension is deprecated and
+// this test suite exists for backwards-compatibility tests only.
 class TestAnnotationRewriting {
 
     @PersistenceUnit(unitName = "test-resource-local")
     private EntityManagerFactory emf;
-    
+
     @PersistenceContext(name = "bogus", unitName = "test")
     private EntityManager em;
 
     @PersistenceContext(unitName = "test", type = PersistenceContextType.EXTENDED)
     private EntityManager extendedEm;
-    
+
+    private String persistenceExtensionEnabledProperty;
+
+    private String jpaExtensionEnabledProperty;
+
     private SeContainer cdiContainer;
 
     TestAnnotationRewriting() {
         super();
     }
-    
+
     @BeforeEach
     void startCdiContainer() {
-        final SeContainerInitializer initializer = SeContainerInitializer.newInstance()
+        this.jpaExtensionEnabledProperty = System.getProperty(JpaExtension.class.getName() + ".enabled");
+        System.setProperty(JpaExtension.class.getName() + ".enabled", "true");
+        this.persistenceExtensionEnabledProperty = System.getProperty(PersistenceExtension.class.getName() + ".enabled");
+        System.setProperty(PersistenceExtension.class.getName() + ".enabled", "false");
+
+        SeContainerInitializer initializer = SeContainerInitializer.newInstance()
             .addBeanClasses(this.getClass());
         assertThat(initializer, notNullValue());
         this.cdiContainer = initializer.initialize();
     }
-  
+
     @AfterEach
     void shutDownCdiContainer() {
         if (this.cdiContainer != null) {
             this.cdiContainer.close();
         }
+        if (this.jpaExtensionEnabledProperty == null) {
+            System.clearProperty(JpaExtension.class.getName() + ".enabled");
+        } else {
+            System.setProperty(JpaExtension.class.getName() + ".enabled", this.jpaExtensionEnabledProperty);
+        }
+        if (this.persistenceExtensionEnabledProperty == null) {
+            System.clearProperty(PersistenceExtension.class.getName() + ".enabled");
+        } else {
+            System.setProperty(PersistenceExtension.class.getName() + ".enabled", this.persistenceExtensionEnabledProperty);
+        }
     }
 
-    private void onShutdown(@Observes @BeforeDestroyed(ApplicationScoped.class) final Object event,
-                            final TransactionManager tm) throws SystemException {
+    private void onShutdown(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event,
+                            TransactionManager tm) throws SystemException {
         // If an assertion fails, or some other error happens in the
         // CDI container, there may be a current transaction that has
         // neither been committed nor rolled back.  Because the
@@ -106,10 +126,10 @@ class TestAnnotationRewriting {
             tm.rollback();
         }
     }
-  
+
     @PersistenceContext(unitName = "test")
-    private void observerMethod(@Observes final TestIsRunning event,
-                                final EntityManager emParameter) {
+    private void observerMethod(@Observes TestIsRunning event,
+                                EntityManager emParameter) {
         assertThat(event, notNullValue());
 
         assertThat(emParameter, notNullValue());
@@ -136,8 +156,8 @@ class TestAnnotationRewriting {
         try {
           em = this.emf.createEntityManager(SynchronizationType.UNSYNCHRONIZED, null);
           fail("Was able to pass a non-null SynchronizationType");
-        } catch (final IllegalStateException expected) {
-          
+        } catch (IllegalStateException expected) {
+
         } finally {
           if (em != null && !em.isOpen()) {
             em.close();
@@ -154,29 +174,28 @@ class TestAnnotationRewriting {
     }
 
     @Test
-    @Disabled // TODO See https://github.com/helidon-io/helidon/issues/8122
     void testNonTransactionalEntityManager() {
         this.cdiContainer.getBeanManager()
             .getEvent()
             .select(TestIsRunning.class)
             .fire(new TestIsRunning("testNonTransactionalEntityManager"));
-        final Set<Annotation> qualifiers = new HashSet<>();
+        Set<Annotation> qualifiers = new HashSet<>();
         qualifiers.add(ContainerManaged.Literal.INSTANCE);
-        qualifiers.add(JpaTransactionScoped.Literal.INSTANCE);
-        final EntityManager entityManager = this.cdiContainer.select(EntityManager.class, qualifiers.toArray(new Annotation[qualifiers.size()])).get();
+        qualifiers.add(JpaTransactionScoped.Literal.INSTANCE); // Note that this is from the old stuff and is deprecated.
+        EntityManager entityManager = this.cdiContainer.select(EntityManager.class, qualifiers.toArray(new Annotation[qualifiers.size()])).get();
         assertThat(entityManager, instanceOf(DelegatingEntityManager.class));
         assertThat(entityManager.isOpen(), is(true));
         assertThat(entityManager.isJoinedToTransaction(), is(false));
         try {
             entityManager.persist(new Object());
             fail("A TransactionRequiredException should have been thrown");
-        } catch (final TransactionRequiredException expected) {
+        } catch (TransactionRequiredException expected) {
 
         }
         try {
             entityManager.close();
             fail("Closed EntityManager; should not have been able to");
-        } catch (final IllegalStateException expected) {
+        } catch (IllegalStateException expected) {
 
         }
     }
@@ -187,12 +206,12 @@ class TestAnnotationRewriting {
             .getEvent()
             .select(TestIsRunning.class)
             .fire(new TestIsRunning("testTransactionalEntityManager"));
-        final Instance<TestAnnotationRewriting> instance = this.cdiContainer.select(TestAnnotationRewriting.class);
-        final TestAnnotationRewriting test = instance.get();
+        Instance<TestAnnotationRewriting> instance = this.cdiContainer.select(TestAnnotationRewriting.class);
+        TestAnnotationRewriting test = instance.get();
         assertThat(test, notNullValue());
         test.testEntityManagerIsJoinedToTransactionInTransactionalAnnotatedMethod();
     }
-    
+
     @Transactional
     void testEntityManagerIsJoinedToTransactionInTransactionalAnnotatedMethod() {
         assertThat(this.em, notNullValue());
@@ -200,22 +219,22 @@ class TestAnnotationRewriting {
         try {
             this.em.close();
             fail("Closed EntityManager; should not have been able to");
-        } catch (final IllegalStateException expected) {
+        } catch (IllegalStateException expected) {
 
         }
     }
 
-    private static final class TestIsRunning {
+    private static class TestIsRunning {
 
         private final String test;
 
-        private TestIsRunning(final String test) {
+        private TestIsRunning(String test) {
             super();
             this.test = test;
         }
 
         @Override
-        public final String toString() {
+        public String toString() {
             return this.test;
         }
     }

--- a/integrations/cdi/jpa-cdi/src/test/java/io/helidon/integrations/cdi/jpa/TestAnnotationRewriting.java
+++ b/integrations/cdi/jpa-cdi/src/test/java/io/helidon/integrations/cdi/jpa/TestAnnotationRewriting.java
@@ -82,10 +82,8 @@ class TestAnnotationRewriting {
 
     @BeforeEach
     void startCdiContainer() {
-        this.jpaExtensionEnabledProperty = System.getProperty(JpaExtension.class.getName() + ".enabled");
-        System.setProperty(JpaExtension.class.getName() + ".enabled", "true");
-        this.persistenceExtensionEnabledProperty = System.getProperty(PersistenceExtension.class.getName() + ".enabled");
-        System.setProperty(PersistenceExtension.class.getName() + ".enabled", "false");
+        this.jpaExtensionEnabledProperty = System.setProperty(JpaExtension.class.getName() + ".enabled", "true");
+        this.persistenceExtensionEnabledProperty = System.setProperty(PersistenceExtension.class.getName() + ".enabled", "false");
 
         SeContainerInitializer initializer = SeContainerInitializer.newInstance()
             .addBeanClasses(this.getClass());

